contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -1954,40 +1977,43 @@ class tsqlBuilder : public tsqlCommonMutator
 
 	void exitChar_string(TSqlParser::Char_stringContext *ctx) override
 	{
-		if (ctx->STRING())
-		{
-			std::string str = ctx->STRING()->getSymbol()->getText();
+		std::string str = getFullText(ctx);			
 
-			if (str.front() == 'N')
-			{
-				// This is only to make the assertion on str.front() easy (below)
-				str.erase(0, 1);	
-			}
+		if (str.front() == 'N')
+		{
+			// This is only to make the assertion on str.front() easy (below)
+			str.erase(0, 1);	
```

Comment:
I just remembered for pure char part erase or replace, if the length of replaced char didn't exceed the original char, then we can do that in enterXXX implements, not in exitXXX implements. Like enterConstant function.

Reply:
We are not actually erasing anything here. This only removes the leading 'N' locally so that the subsequent test is easier -- as the comment on line 1984 indicates (will update the comment to clarify more).